feat(llm-d): integrate vsr with llm-d

srampal opened the pull request:
**What type of PR is this?**
feat(llm-d): Test vLLM semantic Router + LLM-D solution on a single combined gateway. Add a deployment guide and deployment manifests.

**What this PR does / why we need it**:
Validates a solution that combines vLLM Semantic Router and adds deployment guide and example manifests to allow this to be tested.

**Which issue(s) this PR fixes**:
https://github.com/vllm-project/semantic-router/issues/39

Fixes #
Sub-issue #88 within #39

Release Notes: Yes/No

Xunzhuo commented on deploy/kubernetes/llmd-base/README.md:
````diff
@@ -0,0 +1,336 @@
+
+```bash
+# Model name llama3-8b provided explicitly, should route to this backend 
+curl http://192.168.49.2:32293/v1/chat/completions   -H "Content-Type: application/json"   -d '{
````
Can u change the IP?

srampal commented:
> Can you describe the architecture you are adding now? Like where VSR is and where InferencePool is, and how they work together

This design is fully aligned and compatible with the Istio gateway integration feature (before LLM-D) that I pushed in the past. In both cases, I neither use original_dst cluster of Envoy nor InferencePools. InferencePools are only used by the LLM-D/ EPP but not by VSR. Take a look at the httproute manifests. The user adds routes for header match of "x-selected-model" returned by vsr which point to the InferencePool for the corresponding base model.  Also note that endpoint1 IP configured in the VSR config does not matter and is not used (ideally it should be removed from vsr config in future). When I added the Istio gateway integration, I also added the "clear_route_cache" to the vsr config previously for the Istio integration (see deploy/kubernetes/istio/config.yaml). This is used to make the gateway logic work for both basic Istio and Istio + LLM-D cases. Hope that helps. I have tested it out to confirm these steps work correctly.
